pkg/vmprovider/providers/vsphere/virtualmachine/configspec.go

```diff
@@ -80,6 +81,11 @@ func CreateConfigSpec(
 		configSpec.MemoryAllocation.Limit = &lim
 	}
 
+	// Use firmware type from the image if config spec doesn't have it.
+	if configSpec.Firmware == "" && imageFirmware != "" {
```

Comment:
I thought we said the image would override the ConfigSpec since that is what AMIs do?

Reply:
I thought the order would be VMI status.Firmware > configSpec.Firmware > CL setting BIOS/EFI firmware ?. 

meaning we should still let the vi admin specify configSpec.Firmware and we'll honor it?

Reply:
Based on our conversation with the GUI team, they will work with us to ensure the firmware selection is optional in the VM Class Create/Edit workflow in vSphere. This means the above logic is fine, since an admin will be able to omit selecting a firmware.